### PR TITLE
Refactore: randomize sub configs

### PR DESCRIPTION
**Pull request by erfjab**
The method of implementing randomization of configs in the previous method worked, but it was not optimal. I found a better way and implemented it.

**Tasks:**
- [x] **Refactore:** Randomize sub configs

### CODE REVIEW

File: app/subscription/share.py [hunk 1]
```diff
@@ -288,6 +245,8 @@ def process_inbounds_and_tags(
     inbounds = sorted(
         _inbounds, key=lambda x: index_dict.get(x[1][0], float('inf')))
 
+    all_hosts = []
```
[the remark this answers is not in the document]
Comment (erfjab) [reply app/subscription/share.py:248]: https://github.com/Gozargah/Marzban/pull/1144/commits/00f3fdf2a635553b758b474ff025366980f0d84c#r1684463555

[the remark this answers is not in the document]
Comment (erfjab) [reply app/subscription/share.py:248]: I don't think I messed up the code, but if you have a better solution, please provide more details.

**Comment by M03ED** [comment]
i mean this #1146

**Comment by erfjab** [comment]
> i mean this #1146

Thank you for your attention, you are randomizing the inbounds, but I want to randomize the hosts as well.

**Comment by M03ED** [comment]
This pr causes a big problem 
The shuffle function is gonna shuffle all the data inside the dictionary, for example the host of the first host is gonna be used for the third host

**Comment by erfjab** [comment]
> This pr causes a big problem 
> The shuffle function is gonna shuffle all the data inside the dictionary, for example the host of the first host is gonna be used for the third host 

I am testing again, if there is a problem, I will fix it.

**Comment by erfjab** [comment]
ok, i fix this

**Comment by erfjab** [comment]
#1197 ✅

**Pull request merged**